### ISSUE TITLE
Cannot read properties of undefined (reading 'parse')

### ISSUE THREAD

**Issue by ProNotion** [SimplifyJobs/webpack-ext-reloader#417]
[//]: # (These comments will not show up. You can keep them or delete them, up to you.)

I am getting an error when attempting to run my vue project using `vue-cli-service build --mode development`

[//]: # (A clear and concise description of what the bug is.)

#### Error Stacks

```bash
TypeError: Cannot read properties of undefined (reading 'parse')
    at extractEntries ([PATH]\node_modules\webpack-ext-reloader\dist\webpack:\webpack-ext-reloader\src\utils\manifest.ts:12:30)
    at ExtensionReloaderImpl._registerPlugin ([PATH]\node_modules\webpack-ext-reloader\dist\webpack:\webpack-ext-reloader\src\ExtensionReloader.ts:81:23)
    at ExtensionReloaderImpl.apply ([PATH]\node_modules\webpack-ext-reloader\dist\webpack:\webpack-ext-reloader\src\ExtensionReloader.ts:111:12)
```

##### Config

```javascript
const { resolve } = require("path");
const CopyWebpackPlugin = require('copy-webpack-plugin');
const ExtReloader = require('webpack-ext-reloader');

const { defineConfig } = require('@vue/cli-service')
module.exports = defineConfig({
  transpileDependencies: true,
  configureWebpack: {
    entry: {
      background: './src/background.js',
      main: './src/main.js',
    },
    output: {
      filename: (pathData) => {
        return pathData.chunk.name === 'background' ? 'js/background.js' : 'js/[name].[contenthash].js';
      }
    },
    plugins: [
      new ExtReloader({
        manifest: resolve(__dirname, "public/manifest.json")
      }),
      new CopyWebpackPlugin({
        patterns: [
          { from: 'public/manifest.json', to: 'manifest.json' },
          // If you have icons or other assets you need to include, add them here
          { from: 'public/icons', to: 'icons' },
        ],
      }),
    ],
  }
})
```

##### Manifest

```json
{
    "manifest_version": 3,
    "name": "MYEXTENSION",
    "version": "1.0",
    "description": "Extension Description",
    "action": {
      "default_popup": "index.html",
      "default_icon": {
        "16": "icons/get_started16.png",
        "32": "icons/get_started32.png",
        "48": "icons/get_started48.png",
        "128": "icons/get_started128.png"
      }
    },
    "permissions": ["storage", "activeTab", "identity"],
    "background": {
      "service_worker": "js/background.js"
    }
  }
```

[//]: # (A minimal example to reproduce the behavior if relevant.)

#### Expected behavior

I expected it to build

[//]: # (A clear and concise description of what you expected to happen.)

#### Environment:

- OS: `Windows`
- Browser Version:
- webpack Version:
- webpack-ext-reloader Version: `^1.1.10`

#### I'm going to open a PR

- [ ] yes
- [X] no

**Comment by vialoh** [comment]
Your stack trace says `JSON5.parse` is undefined at https://github.com/SimplifyJobs/webpack-ext-reloader/blob/master/src/utils/manifest.ts#L12.

The only reason I can think of for why it would be undefined is if `JSON5` isn't installed (try clearing `node_modules` and/or lock files and reinstalling dependencies) or maybe something with your particular setup is preventing it from being imported correctly.

**Comment by ProNotion** [comment]
Thanks @vialoh I tried removing both and it made no difference. JSON5 is in the node_modules directory,

**Comment by vialoh** [comment]
What does `transpileDependencies` do in `@vue/cli-service`'s `defineConfig`?

**Comment by ProNotion** [comment]
@vialoh It is explained [here](https://cli.vuejs.org/config/#transpiledependencies)

**Comment by vialoh** [comment]
I don't know if the problem is with `webpack-ext-reloader`. My best guess is that there is something strange going on with your setup preventing modules from being imported correctly. If you can provide an example repo with the problem, I'll take a closer look when I have time.

**Comment by vialoh** [comment]
Have you tried it without `transpileDependencies: true`?

**Comment by svchostdotexe** [comment]
i have the exact same issue. tried to clean reinstall on a fresh project but still the same

**Comment by vialoh** [comment]
My best guess is that the issue is not with `webpack-ext-reloader` but with how `@vue/cli-service` transpiles dependencies. I don't have time to dig into this right now but if you can provide an example repo, I'll take a closer look when I have time.

**Comment by ProNotion** [comment]
@vialoh I will see about putting a repo together but even with `transpileDependencies` set to `false` the issue is present.

**Comment by adamtaylor13** [comment]
Adding a strange note here that I'm having the same issue and the stack trace is referring to files that I can't seem to find.

Stack Trace:
```
╰ $ yarn watch
yarn run v1.22.19
$ NODE_ENV=development webpack --config webpack/webpack.dev.js --mode=development --watch
[webpack-cli] TypeError: Cannot read properties of undefined (reading 'parse')
    at extractEntries (/Users/.../node_modules/webpack-ext-reloader/dist/webpack:/webpack-ext-reloader/src/utils/manifest.ts:12:30)
```

But:

```
╭ user@comp:~/.../node_modules/webpack-ext-reloader/dist (main)
╰ $ ll
total 2488
drwxr-xr-x@  4 user  staff   128B Nov  1 15:45 ./
drwxr-xr-x@ 17 user  staff   544B Nov  1 15:44 ../
-rwxr-xr-x@  1 user  staff   628K Nov  1 15:39 webpack-ext-reloader-cli.js*
-rw-r--r--@  1 user  staff   609K Nov  1 15:39 webpack-ext-reloader.js
```

**Comment by mhorlacher** [comment]
Having the exact same issue on a fresh project. Seems the issues is affecting several users at this point.

**Comment by adamtaylor13** [comment]
I wonder if it's worth completely reverting this commit: https://github.com/SimplifyJobs/webpack-ext-reloader/commit/0c4fe869f009788fc8bd2f272cf749c22b6a3247

Since [v3 is on indefinite hold](https://groups.google.com/a/chromium.org/g/chromium-extensions/c/zQ77HkGmK9E)

RFC: https://github.com/SimplifyJobs/webpack-ext-reloader/pull/435

Alternatively, we could investigate why this module doesn't exist. Supposedly `JSON5` should've been installed when installed this module? 🤔

**Comment by vialoh** [comment]
> ```js
> const replaced = myText.replace(/\s+\/\/.*/g, "")
> ```

Correct me if I'm wrong, but I think that regex would remove potentially valid JSON and break the JSON parsing. For example:
```json
{
  "version": 3,
  "foo": " // bar"
}
```

Would become invalid JSON, missing some of the intended string:
```json
{
  "version": 3,
  "foo": "
}
```

As for why someone would have a string like that within their manifest, I don't know, but it should remain unchanged. JSON5 covers all of this so we don't have to worry about it.

**Comment by vialoh** [comment]
Is everyone in this thread using Vue/Vite? If someone can provide an example repo demonstrating the problem, I can probably fix this quickly. Right now I have no idea what the problem is because we're not having any issues with it on our end.

**Comment by adamtaylor13** [comment]
> As for why someone would have a string like that within their manifest, I don't know, but it should remain unchanged. JSON5 covers all of this so we don't have to worry about it.

You are correct regarding how that would mangle the JSON, however—2 counter-points, if I might:

1. We only need the parsing to not choke, just enough to pull out the `background` and `content_scripts` properties. Neither of which can validly contain a `"// "` sequence in them.

2. Considering the scope of this project is so narrow—reload a chrome extension when the project changes—my opinion is that it's better for long-term maintainability to remove dependencies when the solution can be replaced with a single line with very-little room for potential bugs from that.

3. (Bonus opinion you didn't ask for!) Considering at least point 1. above, it may be worth investigating this just to solve this bug for folks ¯\\_(ツ)_/¯

At any rate, I don't feel particularly strongly about this—I assumed this would be a fairly easy approach to solving this issue without many downsides, but I do understand if your reservations re: mangling the JSON make you hesitant to do this. I appreciate you hearing out my opinions even if you don't agree 😛

**Comment by ProNotion** [comment]
@vialoh  Sorry for the delay but I have put together a simplified repo which demonstrates the issues - https://github.com/ProNotion/webpack-ext-reloader-example

**Comment by vialoh** [comment]
@adamtaylor13 I appreciate your feedback. Importing a library shouldn't cause issues, so while removing JSON5 may technically work, it would not solve the root cause of the problem. My guess is that it has something to do with how Vue/Vite transpiles dependencies.

@ProNotion Thank you! I will see what I can do with this today.

**Comment by vialoh** [comment]
@ProNotion (and anyone else) can you verify for me that version 1.1.12 fixes your issue? The problem was indeed with how Vue/Vite imports dependencies. Relevant commit to fix the JSON5 import: cf971cb52e4417ce955d28726b7134e06612ca23

The original entry finding code also did not allow the Webpack config's `output.filename` to be a function, and I think the original filename comparisons were flawed, but that should be fixed now as well: 9fc2a78185034e222166ca84813c02125208ec97

**Comment by ProNotion** [comment]
@vialoh Thank you - I can confirm that the build error has now gone 🙌  Great work!
